### PR TITLE
feat: add config parameter to redeclare same site cookie setup

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
In case of openid redirect the cookie samesite strict or lax are not working.

Use:
```php
<?php
$CONFIG = [
        "http.cookie.samesite" => "None",
```

to choose a different setup

## Related Issue
- Fixes https://github.com/owncloud/enterprise/issues/4191

## How Has This Been Tested?
- http://openid-azure.owncloud.works/

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by mmattel** [comment]
This will be docs relevant, pls do not forget to file a docs issue for a `config-to-docs` run when close to merge.

**Review by C0rby — APPROVED** [review]
Looks good

File: lib/private/Session/CryptoWrapper.php [hunk 1]
```diff
@@ -93,13 +89,14 @@ public function __construct(IConfig $config,
 				if (\version_compare(PHP_VERSION, '7.3.0') === -1) {
```
Comment (IljaN) [inline lib/private/Session/CryptoWrapper.php:89]: Why does this depend on the php version?

Comment (DeepDiver1975) [reply lib/private/Session/CryptoWrapper.php:89]: method signature was changed with 7.3 - read the docs on setcookie

**Comment by DeepDiver1975** [comment]
@micbar pr is ready to be merged and released

**Comment by mmattel** [comment]
@DeepDiver1975 I know you don´t like it, but pls do: https://github.com/owncloud/core/pull/38458#issuecomment-788842045

Can you pls give more info what the values mean/do? `Possible values: Strict, Lax or None`

**Review by micbar — APPROVED** [review]

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 I know you don´t like it, but pls do: [#38458 (comment)](https://github.com/owncloud/core/pull/38458#issuecomment-788842045)

https://github.com/owncloud/docs/issues/3260

**Comment by DeepDiver1975** [comment]
> Can you pls give more info what the values mean/do? `Possible values: Strict, Lax or None`

the internet is full of explanation around this .... no need to duplicate this information.

**Comment by DeepDiver1975** [comment]
> EDIT: let me rephrase, what would be the impact to a non openid environment when using strict or lax compared to the existing setup like without this PR?

the default is 'strict' - same as before this pr - no no change in behavior.

File: config/config.sample.php [hunk 2]
```diff
@@ -265,6 +265,15 @@
  */
 'csrf.disabled' => false,
 
+/**
+ * Allows to relax ownClouds same site cookie settings.
+ *
+ * Possible values: Strict, Lax or None
+ * Setting the same site cookie to none is necessary in case of OpenID Connect.
+ */
```
Comment (mmattel) [inline config/config.sample.php:273]:
```suggestion
 * Define how to relax same site cookie settings for ownCloud
 *
 * Possible values: Strict, Lax or None
 * Setting the same site cookie to None is necessary in case of OpenID Connect.
 * For more information about the impact of the values see:
 * https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite#values
 */
```

**Comment by micbar** [comment]
@mmattel Thanks for the commit. Waiting for CI

**Comment by jnweiger** [comment]
> > Can you pls give more info what the values mean/do? `Possible values: Strict, Lax or None`
> 
> the internet is full of explanation around this .... no need to duplicate this information.

Admin docu should make the connection from the error message "Unable to determine state" to the config setting "http.cookie.samesite"
I doubt the internet knows that much.

**Comment by mmattel** [comment]
I added a link to the docs issue (https://github.com/owncloud/docs/issues/3260#issuecomment-790482935) where there is more information. If you agree we take a closer look on that link if the information provided satisfies the needs and in case not, we have to ask @DeepDiver1975 for an in depth input.

**Comment by gabi18** [comment]
On macOS Mojave 10.14.6 (Safari 12.1.2) the error persists, see
https://github.com/owncloud/core/issues/38519
